lib/gringotts/gateways/pin_pay.ex

```diff
@@ -335,50 +335,38 @@ defmodule Gringotts.Gateways.Pinpay do
   # network request in here, and parse it using another private method called
   # `respond`.
 
-  defp card_params(card, opts) do
-    [
-      "card[number]": card.number,
-      "card[name]": card.first_name <> card.last_name,
-      "card[expiry_month]": card.month |> Integer.to_string() |> String.pad_leading(2, "0"),
-      "card[expiry_year]": card.year |> Integer.to_string(),
-      "card[cvc]": card.verification_code,
-      "card[address_line1]": opts[:address].street1,
-      "card[address_city]": opts[:address].city,
-      "card[address_country]": opts[:address].country
-    ]
-  end
-
   defp card_for_token(card, opts) do
     [
-      "number": card.number,
-      "name": card.first_name <> card.last_name,
-      "expiry_month": card.month |> Integer.to_string() |> String.pad_leading(2, "0"),
-      "expiry_year": card.year |> Integer.to_string(),
-      "cvc": card.verification_code,
-      "address_line1": opts[:Address][:street1],
-      "address_city": opts[:Address][:city],
-      "address_country": opts[:Address][:country]
+      number: card.number,
+      name: CreditCard.full_name(card),
+      expiry_month: card.month |> Integer.to_string() |> String.pad_leading(2, "0"),
+      expiry_year: card.year |> Integer.to_string(),
+      cvc: card.verification_code,
+      address_line1: opts[:Address][:street1],
+      address_city: opts[:Address][:city],
+      address_country: opts[:Address][:country]
     ]
   end
 
-  @spec commit(atom, String.t(), keyword, keyword) :: {:ok | :error, Response}
-  defp commit(:post, endpoint, param, opts) do
-    auth_token = encoded_credentials("c4nxgznanW4XZUaEQhxS6g", "")
+  @spec commit(atom, String.t(), keyword) :: {:ok | :error, Response}
+  defp commit(:post, endpoint, param) do
+    auth_token = encoded_credentials(param[:config].apiKey, param[:config].pass)
```

Comment:
Isn't `param[:config].pass` always `""`? And you cannot rely on the presence of this element because it is not in your `required_config` list.